req_compile/dists.py

```diff
@@ -83,12 +78,12 @@ def extras(self):
                 extras |= set(reason.extras)
         return extras
 
-    def add_reason(self, node, reason):
-        # type: (DependencyNode, Optional[pkg_resources.Requirement]) -> None
+    def add_reason(
+        self, node: "DependencyNode", reason: Optional[pkg_resources.Requirement]
```

Comment:
You can use the real type here, but you might need `from __future__ import annotation`